### DISCUSSION TITLE
UVs, shader assignments, variants

### DISCUSSION THREAD

**Discussion by Xaia** [Autodesk/maya-usd#1548] [Q&A]
Hi there,

I have a few questions since I cannot get this to work correctly.
I am trying to export Mash instancer as USD with UV's - nothing special, simple basic stuff. It looks like USD is not exporting UVs. 

1. Can anyone tell me how to check the UVs in USD in Maya? 

2. Another thing is that for some reason Maya USD sets geometry.arbitrary.map1 for uv set names and Alembic export from Maya geometry.arbitrary.st  which is more universal. How and why is that?
Because of that I cannot read the USD UV sets in Katana directly and have to rename the attributes to make it work. No issues with Alembics.

3. Next thing - How do I assign shaders to a stage prims? There is no procedural way of assigning shaders in Maya yet (like in Katana) and normal assignments are not working.

4. How can I set the variants and purpose (guide, proxy, render) prims? Is there a way without additional scripting?

5. If I add "Def" "scope" or "xform" by mistake, how do I delete it? :)

Cheers

**Comment by santosd** [comment]
Hello @Xaia,

I answered some of your questions on the [beta forums](https://feedback.autodesk.com/project/forum/thread.html?cap=037743fd817049f18d3a9e014e771fd6&forid=%7B0176388e-af2c-46da-8c62-88c0e598e887%7D&topid=%7B22FAF03D-43C2-4C75-9845-9C875111BA15%7D&tp=1&prr=0&a=added_r#Newreply) and included some videos.

In regards to Variants, there currently isn't a workflow implemented and needs additional scripting. This is still something that is in the works on our end.

Unfortunately USD data at this point can't be deleted. If you accidentally add a prim one way is to undo the creation of the prim, but attempting to delete it will just deactivate it. This is done to protect the USD data, but if you have workflow in which you would want USD data completely deleted, please let us know and I can present the information to the team.

Cheers,

David Santos

**Comment by santosd** [comment]
More information on deleting of a prim can be found here: https://github.com/Autodesk/maya-usd/issues/1538#issuecomment-875033046

**Comment by Xaia** [comment]
Hey @santosd 

I am looking to the variants and purpose support very much!

Lets say I create an anonymous layer and by mistake I create def instead of xform, what then? I really don't want to have thousands left overs because an artist click wrong, calculated wrong or just thought it will be needed but it was not in the end. 
We need to be able to delete wherever we created otherwise there will be a mess. 
Redoing and rearranging groups and objects is part of pipeline. 

Cheers
Xaia
